### PR TITLE
WIP:: [CONTRACTS] Debt Token implementation

### PR DESCRIPTION
**Pull request by adibas03**

### CODE REVIEW

**Comment by adibas03** [comment]
@tspoff Do we need to add the Debt Registry?
I can't seem to wrap my head around, how it's role goes

File: ethereum/contracts/loans/student-loans/StudentLoanCrowdfund.sol [hunk 1]
```diff
@@ -51,30 +53,64 @@ contract StudentLoanCrowdfund is ICrowdfund, Ownable {
...
         emit DebtTokenSet(_debtTokenAddr);
     }
 
-    function addFunding(uint amount) public returns (uint tokenId) {
+    function addFunding(uint amount) public belowMaxSupply(amount) returns (uint tokenId) {
         //Calculate how many debt tokens to give them based on loan data
+        // Allocate exact amount
+        StudentLoanLibrary.StoredParams memory params;
+        (
+          params.principalTokenIndex,
```
Comment (adibas03) [inline ethereum/contracts/loans/student-loans/StudentLoanCrowdfund.sol]: @tspoff I see us needing another parameter field, for the crowdfund period.
To ensure the crowdfund does not go on forever.

**Comment by adibas03** [comment]
@tspoff  
adding in the crowdfundLength to termStorage, mean we will be hitting stack errors,
so I am thinking of having it only within the crowdfund contract, instead of sending it to the termStorage.
Do you have a particular concern with that?

**Comment by adibas03** [comment]
Also,
Do we have any plan to cater to funds accidentally sent to the crowdfund address, without going through the contract or things like the airdrops.
Never mind, on second thought, it sounds pointless,.

**Comment by tspoff** [comment]
I think it's fine if we have the length on the crowdfund contract. 

Accidental funds? Well we can have the fallback function reject ether - now that you mention it, that should definitely be explicitly defined.

Yea for airdrops we can't do much - people can always send ERC20's to an address.

**Comment by adibas03** [comment]
Moving this to the new contracts Repo 
https://github.com/enabledao/enable-contracts/pull/2

**Pull request closed without merging**